src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/LibGit2SharpHelpers.cs

```diff
@@ -91,6 +91,25 @@ internal static void SafeCheckout(Repository repo, string commit, CheckoutOption
             }
         }
 
+        /// <summary>
+        /// Adds a file to the repo's index respecting the original file's mode.
+        /// </summary>
+        /// <param name="repo">Repo to add the files to</param>
+        /// <param name="file">Original GitFile to add</param>
+        /// <param name="fullPath">Final path for the file to be added</param>
+        /// <param name="log">Logger</param>
+        internal static void AddFileToIndex(Repository repo, GitFile file, string fullPath, ILogger log)
+        {
+            var fileMode = (Mode)Convert.ToInt32(file.Mode, 8);
```

Comment:
You put this same code in the previous PR. Can this be moved into a helper method?